File: app/src/test/java/com/dpis/module/MainActivityConfiguredCountTest.java
package com.dpis.module;

import org.junit.Test;

import java.util.Set;

import static org.junit.Assert.assertEquals;

public class MainActivityConfiguredCountTest {
    @Test
    public void configuredCountIncludesSavedConfigAndKnownScopeOnly() {
        FakePrefs prefs = new FakePrefs();
        prefs.edit()
                .putInt("viewport.com.example.saved.width_dp", 360)
                .commit();
        DpisConfigStore store = new DpisConfigStore(prefs);

        assertEquals(2, MainActivity.countUserVisibleConfiguredPackages(
                store,
                new MainActivity.ScopeState(Set.of(
                        "com.example.injected",
                        "com.example.saved"), true)));
    }

    @Test
    public void configuredCountDoesNotInferUnknownLegacyScope() {
        assertEquals(0, MainActivity.countUserVisibleConfiguredPackages(
                null,
                new MainActivity.ScopeState(Set.of("com.example.legacy"), false)));
    }

    @Test
    public void configuredCountExcludesSystemFrameworkScopeAliases() {
        assertEquals(1, MainActivity.countUserVisibleConfiguredPackages(
                null,
                new MainActivity.ScopeState(Set.of(
                        "system",
                        "android",
                        "com.example.injected"), true)));
    }

    @Test
    public void configuredCountMatchesKnownScopeOnlyPackage() {
        assertEquals(1, MainActivity.countUserVisibleConfiguredPackages(
                null,
                new MainActivity.ScopeState(Set.of("com.example.injected"), true)));
    }
}

File: app/src/test/java/com/dpis/module/shell/MainActivityConfiguredCountTest.kt
package com.dpis.module

import com.dpis.module.applist.ScopeState
import com.dpis.module.config.DpisConfigStore
import org.junit.Assert.assertEquals
import org.junit.Test

class MainActivityConfiguredCountTest {
    @Test
    fun configuredCountIncludesSavedConfigAndKnownScopeOnly() {
        val prefs = FakePrefs()
        prefs.edit()
            .putInt("viewport.com.example.saved.width_dp", 360)
            .commit()
        val store = DpisConfigStore(prefs)

        assertEquals(
            2,
            MainActivity.countUserVisibleConfiguredPackages(
                store,
                ScopeState(
                    setOf("com.example.injected", "com.example.saved"),
                    true,
                ),
            ),
        )
    }

    @Test
    fun configuredCountDoesNotInferUnknownLegacyScope() {
        assertEquals(
            0,
            MainActivity.countUserVisibleConfiguredPackages(
                null,
                ScopeState(setOf("com.example.legacy"), false),
            ),
        )
    }

    @Test
    fun configuredCountExcludesSystemFrameworkScopeAliases() {
        assertEquals(
            1,
            MainActivity.countUserVisibleConfiguredPackages(
                null,
                ScopeState(
                    setOf("system", "android", "com.example.injected"),
                    true,
                ),
            ),
        )
    }

    @Test
    fun configuredCountMatchesKnownScopeOnlyPackage() {
        assertEquals(
            1,
            MainActivity.countUserVisibleConfiguredPackages(
                null,
                ScopeState(setOf("com.example.injected"), true),
            ),
        )
    }
}
